tests/ssl-opt.sh

```diff
@@ -2830,6 +2830,7 @@ run_test    "Authentication: send CA list in CertificateRequest, client self sig
 
 # Tests for certificate selection based on SHA verson
 
+requires_config_enabled MBEDTLS_SSL_KEEP_PEER_CERTIFICATE
```

Comment:
Many tests get disabled here for the "non-keep" builds, whereas some unit tests (in test suites) have been added, but in smaller quantity. How does this impact the test coverage in the "non-keep" case? Is it equally tested, or, if it is difficult in the current setup (the way the client and server are implemented), maybe we should at least plan adding more tests, potentially involving client and server modifications?

Reply:
@k-stachowiak I've re-introduced all the tests that were previously disabled, by getting the CRT info the tests rely on from a verify callback instead of a call to `mbedtls_ssl_get_peer_cert()` at the end of the handshake.